src/mesh/Router.cpp

```diff
@@ -355,13 +355,21 @@ ErrorCode Router::send(meshtastic_MeshPacket *p)
         if (moduleConfig.mqtt.enabled && isFromUs(p) && mqtt) {
             mqtt->onSend(*p, *p_decoded, chIndex);
         }
+#endif
+#if HAS_UDP_MULTICAST
+        // Only publish to UDP if we're the original transmitter of the packet
+        if (udpHandler && config.network.enabled_protocols & meshtastic_Config_NetworkConfig_ProtocolFlags_UDP_BROADCAST &&
+            isFromUs(p)) {
```

Comment:
`isFromUs(p)` is new here and causes packets from other nodes not to be broadcasted via UDP. 
For MQTT this is here because there is also an `mqtt->onSend()` in the `Router::handleReceived()` to allow for messages you decide not to rebroadcast to also end up on MQTT. If we want that for UDP as well, we'd need it there also, and guard it there with `!isFromUs()`.
If we only want to broadcast packets we'd also broadcast over LoRa, then we'd need to keep it as is and remove the `isFromUs(p)` here.